src/ProjectOrigin.Registry.IntegrationTests/PerformanceTests.cs

```diff
@@ -120,7 +114,90 @@ public async Task TestThroughput()
         var requestsPerSecond = completed.Count / elapsedSeconds;
 
         Console.WriteLine($"Completed {completed.Count} requests in {elapsedSeconds} seconds ({requestsPerSecond} requests per second).");
-        requestsPerSecond.Should().BeGreaterThan(8); // based on througput test on github ~10
+        requestsPerSecond.Should().BeGreaterThan(60); // based on througput test on github ~60
+    }
+
+    [Fact]
+    public async Task TestLatencySequential()
+    {
+        Console.WriteLine($"-- Starting sequential test --");
+
+        int count = 10;
+
+        using var channel = CreateRegistryChannel();
+
+        List<long> measurements = new List<long>();
+        var stopwatch = new Stopwatch();
+
+        foreach (var i in Enumerable.Range(0, count))
+        {
+            stopwatch.Restart();
+            var transaction = await SendRequest(channel).ConfigureAwait(false);
+            while (!await IsCommitted(channel, transaction).ConfigureAwait(false))
+            {
+                await Task.Delay(25).ConfigureAwait(false);
+            }
+
+            stopwatch.Stop();
+            measurements.Add(stopwatch.ElapsedMilliseconds);
+        }
+
+        var ms95th = measurements.OrderBy(x => x).ElementAt((int)(count * 0.95));
+        Console.WriteLine($"Unit: Millisecond delay from published to committed");
+        Console.WriteLine("Max:  " + measurements.Max());
+        Console.WriteLine("95th: " + ms95th);
+        Console.WriteLine("Mean: " + Math.Round(measurements.Average()));
+        Console.WriteLine("Min:  " + measurements.Min());
+        Console.WriteLine($"-- Finished sequential test --");
+
+        ms95th.Should().BeLessThan(6000);
+    }
+
+    [Fact]
+    public async Task TestLatencyParallel()
+    {
+        Console.WriteLine($"-- Starting parallel test --");
+        int count = 100;
+
+        using var channel = CreateRegistryChannel();
+
+        List<long> measurements = new List<long>();
+
+        ConcurrentQueue<(Stopwatch stopwatch, Registry.V1.Transaction transaction)> queued = new();
+
+        for (int i = 0; i < count; i++)
+        {
+            var stopwatch = new Stopwatch();
+            stopwatch.Start();
+            var transaction = await SendRequest(channel).ConfigureAwait(false);
+
+            queued.Enqueue((stopwatch, transaction));
+        }
```

Comment:
It might be, but it doesn't wait for them to be comitted, it worked great to highligt the issue with the JobConsumer where when all the jobs came at the same time, they would all be waiting 30 seconds for the job-slot which made the low throughput.